ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/QuestionAnsweringModel.java

```diff
@@ -28,46 +28,110 @@
 import ai.djl.translate.TranslatorFactory;
 import lombok.extern.log4j.Log4j2;
 
+/**
+ * Question answering model implementation that supports both standard QA and
+ * highlighting sentence.
+ */
 @Log4j2
 @Function(FunctionName.QUESTION_ANSWERING)
 public class QuestionAnsweringModel extends DLModel {
+    // Flag to track highlighting type
+    private String highlightingType = null;
 
     @Override
     public void warmUp(Predictor predictor, String modelId, MLModelConfig modelConfig) throws TranslateException {
-        String question = "How is the weather?";
-        String context = "The weather is nice, it is beautiful day.";
+        // Check highlighting type based on model config
+        checkHighlightingType(modelConfig);
+
         Input input = new Input();
-        input.add(question);
-        input.add(context);
+        input.add(DEFAULT_WARMUP_QUESTION);
+        input.add(DEFAULT_WARMUP_CONTEXT);
 
         // First request takes longer time. Predict once to warm up model.
         predictor.predict(input);
     }
 
+    /**
+     * Checks and sets the highlighting type based on model configuration.
+     *
+     * @param modelConfig The model configuration
+     */
+    private void checkHighlightingType(MLModelConfig modelConfig) {
+        if (modelConfig != null) {
+            if (modelConfig.getModelType() != null) {
+                String modelType = modelConfig.getModelType();
+                if (SENTENCE_HIGHLIGHTING_TYPE.equalsIgnoreCase(modelType)) {
+                    highlightingType = SENTENCE_HIGHLIGHTING_TYPE;
+                }
+            }
+        }
+    }
```

Comment:
You don't need this function and the variable `highlightingType` neither. You can directly compare `SENTENCE_HIGHLIGHTING_TYPE` with `modelConfig.getModelType()` by String member function `equalsIgnoreCase`. For example,
```
SENTENCE_HIGHLIGHTING_TYPE.equalsIgnoreCase(modelConfig.getModelType())
```

Reply:
Updated